src/js/cast/mediamanager.js

```diff
@@ -56,49 +56,50 @@ export const ERROR_TIMEOUT = 5000;
  */
-export default function JWMediaManager(receiverManager, container, events, analyticsConfig) {
-
-    // The current JW Player instance.
-    // TODO: make this an object, which allows for multiple playerInstances per mediaSessionId
-    // so we can do preloading.
-    let playerInstance;
-
-    // Represents the status of a media session.
-    let mediaStatus;
-
-    /**
-     * By default we do not serialize the entire media status.
-     * The following elements should be set in this block if they
-     * should be serialized.
-     *
-     * This object is used to set the flags for which elements in the media
-     * status should be serialized.
-     */
-        // let mediaStatusFlags = [];
-
-        // The messageBus used for sending/receiving messages.
-    let messageBus = receiverManager.getCastMessageBus(cast.receiver.media.MEDIA_NAMESPACE);
-
-    // The requestId that will be associated with the next status update.
-    // So that senders can correlate their actions with a status update.
-    let currentRequestId = 0;
-
-    // Whether we are currently processing a load request.
-    let isLoading = false;
-
-    // The last time an ad pod was initiated.
-    let adPodStartTime = 0;
-
-    // Start listening for messages.
-    messageBus.addEventListener('message', dispatchEvent);
+export default class JWMediaManager {
+    constructor(receiverManager, container, events, analyticsConfig) {
+        this.receiverManager = receiverManager;
+        this.playerManager = this.receiverManager.getPlayerManager();
+        this.container = container;
+        this.events = events;
+        this.analyticsConfig = analyticsConfig;
+        // The current JW Player instance.
+        // TODO: make this an object, which allows for multiple playerInstances per mediaSessionId
+        // so we can do preloading.
+        this.playerInstance;
+        // Represents the status of a media session.
+        this.mediaStatus;
+        /**
+         * By default we do not serialize the entire media status.
+         * The following elements should be set in this block if theycast.receiver.media.MediaStatus
+         * should be serialized.
+         *
+         * This object is used to set the flags for which elements in the media
+         * status should be serialized.
+         */
+            // let mediaStatusFlags = [];
+
+        // The requestId that will be associated with the next status update.
+        // So that senders can correlate their actions with a status update.
+        this.currentRequestId = 0;
+        // Whether we are currently processing a load request.
+        this.isLoading = false;
+        // The last time an ad pod was initiated.
+        this.adPodStartTime = 0;
+
+        // Start listening for messages.
+        // https://developers.google.com/cast/docs/web_receiver/core_features following CAF changes
+        this.playerManager.addEventListener(EventType.ALL, this.dispatchEvent.bind(this));
```

Comment:
was setting this to ALL for debugging but we should listen to specific events for the actual app.